src/utils/specs.py

```diff
@@ -206,30 +301,41 @@ def get_machine_specs(devicenum):
             sys.exit(1)
 
     (
-        gpu_id,
+        gpu_name,
+        gpu_arch,
         L1,
         L2,
-        sclk,
+        max_sclk,
         num_CU,
         num_SIMD,
         num_SE,
         wave_size,
         grp_size,
-        wave_occu,
+        max_waves_per_cu,
+        L2Banks,
+        LDSBanks,
+        numSQC,
+        compute_partition,
+        memory_partition,
     ) = gpuinfo()
+
     rocm_smi = run(["rocm-smi"])
 
     device = rf"^\s*{devicenum}(.*)"
 
     hostname = socket.gethostname()
-    cpu = search(r"^model name\s*: (.*?)$", cpuinfo)
-    kernel = search(r"version (\S*)", version)
+    sbios = (
+        path("/sys/class/dmi/id/bios_vendor").read_text().strip()
```

Comment:
Is this path 100% universal across Linux distros?  I don't know, but if not should consider having  some basic fill error checking here.

Reply:
Checked this across all supported Linux distros and some HPC systems. Looks to always be there.